lib/jsdom/browser/parse5-adapter-parsing.js

```diff
@@ -49,6 +90,14 @@ module.exports = class JSDOMParse5Adapter {
   }
 
   setTemplateContent(templateElement, contentFragment) {
+    // Internally during the construction of the HTMLTemplateElement, a DocumentFragment is also created. At this point
+    // we have 2 fragments the one created in the HTMLTemplateElement constructor and the one created by
+    // createDocumentFragment. We now have to reconciliate the one created internally with one created by the tree
+    // adapter.
```

Comment:
It'd be good to expand this comment to explain why we create one in the HTMLTemplateElement constructor. Would it be accurate to change it to something like "...a DocumentFragment is also created. (This is important for cases like `document.createElement("template")`, where `setTemplateContent` never runs.)

Also for "the one created by createDocumentFragment", do you mean the one created by parse5?

I think it's important to make this comment as explicit as possible as it's kind of a weird situation.

Reply:
> Also for "the one created by createDocumentFragment", do you mean the one created by parse5?

Yes. Will update the comment.